File: native-windows/src/SeleneNative.Core/Models/LunaFeatureModels.cs
using System.Text.Json.Serialization;

namespace SeleneNative.Core.Models;

public sealed class MediaPlatformItem
{
    public string Id { get; init; } = string.Empty;
    public string Title { get; init; } = string.Empty;
    public string Cover { get; init; } = string.Empty;
    public string Author { get; init; } = string.Empty;
    public string Description { get; init; } = string.Empty;
    public string Duration { get; init; } = string.Empty;
    public string Source { get; init; } = string.Empty;
    public string Url { get; init; } = string.Empty;
}

public sealed class TmdbBackdrop
{
    [JsonPropertyName("backdrop")]
    public string? Backdrop { get; init; }

    [JsonPropertyName("poster")]
    public string? Poster { get; init; }

    [JsonPropertyName("logo")]
    public string? Logo { get; init; }

    [JsonPropertyName("title")]
    public string? Title { get; init; }

    [JsonPropertyName("overview")]
    public string? Overview { get; init; }

    [JsonPropertyName("rating")]
    public double? Rating { get; init; }

    [JsonPropertyName("year")]
    public string? Year { get; init; }

    [JsonPropertyName("numberOfSeasons")]
    public int? NumberOfSeasons { get; init; }
}

public sealed class DoubanQuickInfo
{
    public string Title { get; init; } = string.Empty;
    public string? Year { get; init; }
    public string? Rating { get; init; }
    public string? Summary { get; init; }
    public IReadOnlyList<string> Genres { get; init; } = [];
    public IReadOnlyList<string> Directors { get; init; } = [];
    public IReadOnlyList<string> Cast { get; init; } = [];
}

public sealed class DoubanComment
{
    public string Author { get; init; } = string.Empty;
    public string Content { get; init; } = string.Empty;
    public string Rating { get; init; } = string.Empty;
}

public sealed class DoubanRecommendation
{
    public string Id { get; init; } = string.Empty;
    public string Title { get; init; } = string.Empty;
    public string Cover { get; init; } = string.Empty;
    public string Rating { get; init; } = string.Empty;
}

File: native-macos/Sources/SeleneNative/Models/LunaFeatureModels.swift
import Foundation

struct MediaPlatformItem: Codable, Identifiable, Hashable {
    let id: String
    let title: String
    let cover: String
    let author: String
    let description: String
    let duration: String
    let source: String
    let url: String

    init(
        id: String = "",
        title: String = "",
        cover: String = "",
        author: String = "",
        description: String = "",
        duration: String = "",
        source: String = "",
        url: String = ""
    ) {
        self.id = id
        self.title = title
        self.cover = cover
        self.author = author
        self.description = description
        self.duration = duration
        self.source = source
        self.url = url
    }
}

struct TMDBBackdrop: Codable, Hashable {
    let backdrop: String?
    let poster: String?
    let logo: String?
    let title: String?
    let overview: String?
    let rating: Double?
    let year: String?
    let numberOfSeasons: Int?
}

struct DoubanQuickInfo: Hashable {
    let title: String
    let year: String?
    let rating: String?
    let summary: String?
    let genres: [String]
    let directors: [String]
    let cast: [String]
}

struct DoubanComment: Hashable, Identifiable {
    var id: String { "\(author)-\(content)" }
    let author: String
    let content: String
    let rating: String
}

struct DoubanRecommendation: Hashable, Identifiable {
    let id: String
    let title: String
    let cover: String
    let rating: String
}
